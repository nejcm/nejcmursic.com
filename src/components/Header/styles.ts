import styled from '@emotion/styled';
import theme from '../../../config/theme';

export const Wrapper = styled.header`
  > .top {
    position: relative;
    z-index: 100;
    background-color: ${(props: {background?: string; theme: {dark: number}}) =>
      props.theme.dark
        ? theme.colors.bg_dark_color_light
        : props.background
        ? props.background
        : theme.colors.primary_light};
  }

  .nav-bar {
    padding: 1.5rem 0;
    border-bottom: solid 1px transparent;
    transition: padding 0.25s ease-in-out, background-color 0.25s ease-in-out;
  }

  .headroom--pinned .nav-bar {
    background-color: ${(props: {theme: {dark: number}}) =>
      props.theme.dark
        ? theme.colors.bg_dark_color_lighter
        : theme.colors.white};
    padding: 0.6rem 0;
    border-bottom-color: rgba(0, 0, 0, 0.08);
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.08);
  }

  > svg {
    width: 100%;
    height: auto;
    fill: ${(props: {theme: {dark: number}}) =>
      props.theme.dark
        ? theme.colors.bg_dark_color_light
        : theme.colors.primary_light};
    margin-top: -1px;
    vertical-align: top;
  }
`;

export const Flex = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  .logo {
    margin-right: 1.5rem;
  }

  .left {
    display: flex;
    align-items: center;
  }

  .menu {
    a {
      display: inline-block;
      padding: 8px 15px;
      margin-left: 8px;
      font-size: 1.05rem;
      font-weight: bold;
      letter-spacing: 0.2px;
    }
  }

  .right {
    text-align: right;
  }

  .theme-toggler {
    font-size: 1.3rem;
    line-height: 1;
    padding: 5px 10px;
    margin-left: 8px;

    svg {
      transition: all 0.4s ease 0.1s;

      &.dark {
        transform: rotate(40deg);
        mask circle {
          cx: 11;
          cy: 3;
        }
        g circle {
          transform: scale(0);
        }
        > circle {
          r: 8;
        }
      }

      &.light {
        transform: rotate(90deg);
        mask circle {
          cx: 26;
          cy: 1;
        }
        g circle {
          transform: scale(1);
        }
        > circle {
          r: 5;
        }
      }

      > circle,
      g circle {
        transition: all 0.4s ease-in-out;
        transform-origin: center center 0px;
        fill: ${(props: {theme: {dark: number}}) =>
          props.theme.dark
            ? theme.colors.body_dark_color
            : theme.colors.body_color};
      }

      &:hover {
        opacity: 0.75;
      }
    }
  }
`;

export const Title = styled.div`
  padding: 2.5rem 0 1rem 0;
  background-color: ${(props: {theme: {dark: number}}) =>
    props.theme.dark
      ? theme.colors.bg_dark_color_light
      : theme.colors.primary_light};

  h1 {
    margin: 0;
    opacity: 0.9;
  }
`;
