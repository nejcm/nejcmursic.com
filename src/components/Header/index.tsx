import {Link} from 'gatsby';
import React, {ReactNode} from 'react';
import Headroom from 'react-headroom';
import theme from '../../../config/theme';
import website from '../../../config/website';
import logo from '../../../static/images/logo_square.png';
import Container from '../Container';
import {Wave} from '../Icons';
import {useTheme} from '../Theme';
import {Flex, Title, Wrapper} from './styles';

export interface HeaderProps extends React.HTMLAttributes<HTMLElement> {
  headerTitle?: ReactNode;
  headerBackground?: string;
}

const Header: React.SFC<HeaderProps> = ({
  headerTitle,
  headerBackground,
  ...rest
}) => {
  const {dark, toggle} = useTheme();
  const switchTitle = dark ? 'Deactivate dark mode' : 'Activate dark mode';

  return (
    <Wrapper background={headerBackground} {...rest}>
      <div className="top">
        <Headroom>
          <div className="nav-bar">
            <Container size={Container.sizes.lg}>
              <Flex>
                <div className="left">
                  <Link to="/">
                    <img
                      className="logo"
                      src={logo}
                      width="auto"
                      height={48}
                      alt={website.siteTitle}
                    />
                  </Link>
                </div>
                <div className="right menu">
                  {/* <Link to={'/blog'} className="inverse">
                  Blog
                </Link> */}
                  <a
                    href={`/${website.cv}`}
                    target="_blank"
                    className="inverse"
                    rel="nofollow noopener noreferrer"
                  >
                    CV
                  </a>
                  <button
                    className="none theme-toggler"
                    aria-label={switchTitle}
                    title={switchTitle}
                    onClick={() => toggle()}
                  >
                    <svg
                      xmlns="http://www.w3.org/2000/svg"
                      className={dark ? 'dark' : 'light'}
                      viewBox="0 0 20 20"
                      width="1em"
                      height="1em"
                    >
                      <mask id="moon-mask-nool">
                        <rect fill="#fff" x="0" y="0" width="20" height="20" />
                        <circle
                          fill={theme.colors.bg_dark_color}
                          cx="26"
                          cy="1"
                          r="8"
                        />
                      </mask>
                      <circle
                        mask='url("#moon-mask-nool")'
                        cx="10"
                        cy="10"
                        r="5"
                      />
                      <g>
                        <circle cx="18" cy="10" r="1.5" />
                        <circle cx="14" cy="16.9282" r="1.5" />
                        <circle cx="6" cy="16.9282" r="1.5" />
                        <circle cx="2" cy="10" r="1.5" />
                        <circle cx="6" cy="3.0718" r="1.5" />
                        <circle cx="14" cy="3.0718" r="1.5" />
                      </g>
                    </svg>
                  </button>
                </div>
              </Flex>
            </Container>
          </div>
        </Headroom>
      </div>
      {headerTitle ? (
        <>
          <Title>
            <Container size={Container.sizes.lg}>{headerTitle}</Container>
          </Title>
          <Wave />
        </>
      ) : null}
    </Wrapper>
  );
};

export default Header;
