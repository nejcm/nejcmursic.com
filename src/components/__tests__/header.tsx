import {render} from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header headerTitle="Test title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains headerTitle', () => {
    const headerTitle = `Hello World`;
    const {getByText} = render(<Header headerTitle={headerTitle} />);

    const title = getByText(headerTitle);

    expect(title).toBeInTheDocument();
  });
});
