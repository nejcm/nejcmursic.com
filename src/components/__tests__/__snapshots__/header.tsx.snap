// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="css-1alc7ys"
>
  <div
    className="top"
  >
    <div
      className="headroom-wrapper"
      style={
        Object {
          "height": null,
        }
      }
    >
      <div
        className="headroom headroom--unfixed"
        style={
          Object {
            "MsTransform": "translate3D(0, 0, 0)",
            "WebkitTransform": "translate3D(0, 0, 0)",
            "left": 0,
            "position": "relative",
            "right": 0,
            "top": 0,
            "transform": "translate3D(0, 0, 0)",
            "zIndex": 1,
          }
        }
      >
        <div
          className="nav-bar"
        >
          <div
            className="css-1pfi013"
            style={
              Object {
                "maxWidth": "1200px",
              }
            }
          >
            <div
              className="css-15jlfmr"
            >
              <div
                className="left"
              >
                <a
                  href="/"
                >
                  <img
                    alt="Nejc Mursic"
                    className="logo"
                    height={40}
                    src="test-file-stub"
                    width="auto"
                  />
                </a>
              </div>
              <div
                className="right menu"
              >
                <a
                  className="inverse"
                  href="/Nejc_Mursic_CV.pdf"
                  rel="nofollow noopener noreferrer"
                  target="_blank"
                >
                  CV
                </a>
                <button
                  aria-label="Activate dark mode"
                  className="none theme-toggler"
                  onClick={[Function]}
                  title="Activate dark mode"
                >
                  <svg
                    className="light"
                    height="1em"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      id="moon-mask-nool"
                    >
                      <rect
                        fill="#fff"
                        height="20"
                        width="20"
                        x="0"
                        y="0"
                      />
                      <circle
                        cx="26"
                        cy="1"
                        fill="#020B19"
                        r="8"
                      />
                    </mask>
                    <circle
                      cx="10"
                      cy="10"
                      mask="url(\\"#moon-mask-nool\\")"
                      r="5"
                    />
                    <g>
                      <circle
                        cx="18"
                        cy="10"
                        r="1.5"
                      />
                      <circle
                        cx="14"
                        cy="16.9282"
                        r="1.5"
                      />
                      <circle
                        cx="6"
                        cy="16.9282"
                        r="1.5"
                      />
                      <circle
                        cx="2"
                        cy="10"
                        r="1.5"
                      />
                      <circle
                        cx="6"
                        cy="3.0718"
                        r="1.5"
                      />
                      <circle
                        cx="14"
                        cy="3.0718"
                        r="1.5"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-lfsxqt"
  >
    <div
      className="css-1pfi013"
      style={
        Object {
          "maxWidth": "1200px",
        }
      }
    >
      Test title
    </div>
  </div>
  <svg
    height="35"
    viewBox="0 0 512 35"
    width="512"
  >
    <path
      d="M0,35L512,0.2l0,0H0V35z"
    />
  </svg>
</header>
`;
